jclklib/client/notification_msg.cpp

```diff
@@ -104,6 +104,13 @@ PROCESS_MESSAGE_TYPE(ClientNotificationMessage::processMessage)
 
         ClientState *currentClientState = *it;
 
+        struct timespec last_notification_time;
+
+        if (clock_gettime(CLOCK_MONOTONIC, &last_notification_time) == -1)
+            PrintDebug("ClientNotificationMessage::processMessage clock_gettime failed.\n");
+
+        currentClientState->set_last_notification_time(last_notification_time);
```

Comment:
set the time only if clock_gettime() is successful, which mean that you need to put this line in "else"